sql/handler.cc

```diff
@@ -4676,6 +4676,17 @@ void handler::print_error(int error, myf errflag)
   case HA_ERR_PARTITION_LIST:
     my_error(ER_VERS_NOT_ALLOWED, errflag, table->s->db.str, table->s->table_name.str);
     DBUG_VOID_RETURN;
+  case HA_ERR_ROLLBACK:
+    /* Crash if we run with --debug-assert-on-error */
+    DBUG_ASSERT(!debug_assert_if_crashed_table);
+    SET_FATAL_ERROR;
+    my_printf_error(HA_ERR_ROLLBACK,
+                    "Transaction was rolled back internally for table %`s.%`s. "
+                    "This error is wrong non fatal error that should be "
+                    "reported so it can get fixed.",
```

Comment:
the text cannot say "non fatal" after `SET_FATAL_ERROR`. In fact, there is already `ER_ROLLBACK_ONLY` which works fine here, so this can be changed to
```c++
case HA_ERR_ROLLBACK:
  SET_FATAL_ERROR;
  textno= ER_ROLLBACK;
  break;
```

Reply:
840d19e4a60edec79c6ab6e7275c8bfedabfc985 addresses this.